python/ray/data/_internal/execution/streaming_executor_state.py

```diff
@@ -424,17 +424,23 @@ def to_bundle(resource: ExecutionResources) -> Dict:
             req["GPU"] = math.ceil(resource.gpu)
         return req
 
+    total_cpu_usage = 0
     for op, state in topology.items():
         per_task_resource = op.incremental_resource_usage()
         task_bundle = to_bundle(per_task_resource)
         resource_request.extend([task_bundle] * op.num_active_work_refs())
+        if "CPU" in task_bundle:
+            total_cpu_usage += task_bundle["CPU"] * op.num_active_work_refs()
         # Only include incremental resource usage for ops that are ready for
         # dispatch.
         if state.num_queued() > 0:
             # TODO(Clark): Scale up more aggressively by adding incremental resource
             # usage for more than one bundle in the queue for this op?
             resource_request.append(task_bundle)
 
+    total = ray.cluster_resources()
+    if not ("CPU" in total and total_cpu_usage > 0.6 * total["CPU"]):
```

Comment:
Hmm this shouldn't be needed right?

Reply:
Yeah we probably don't need to have this extra condition.